### PR TITLE
Docs: deletes forward auth ref, nginx, traefik guides

### PR DESCRIPTION
**Pull request by ZPain8464**
Removes Forward Auth reference and the NGINX and Traefik guides. These guides rely on Forward Auth capabilities to authorize requests.

### CODE REVIEW

**Review by desimone — APPROVED** [review]

**Review by wasaga — CHANGES_REQUESTED** [review]
redirects need be added 

/docs/guides/nginx.mdx => https://0-20-0.docs.pomerium.com/docs/guides/nginx.mdx

File: static/_redirects [hunk 1]
```diff
@@ -275,3 +275,7 @@ https://0-12-0.docs.pomerium.io/ https://0-12-0.docs.pomerium.com/:splat 301!
 https://0-11-0.docs.pomerium.io/ https://0-11-0.docs.pomerium.com/:splat 301!
 https://0-10-0.docs.pomerium.io/ https://0-10-0.docs.pomerium.com/:splat 301!
 
+# 301 deprecated guides in v21 to v20 guides
+https://0-21-0.docs.pomerium.io/docs/guides/nginx https://0-20-0.docs.pomerium.com/docs/guides/nginx 301!
+https://0-21-0.docs.pomerium.io/docs/guides/traefik-ingress https://0-20-0.docs.pomerium.com/docs/guides/traefik-ingress 301!
```
Comment (wasaga) [inline static/_redirects:280]: maybe it should be 

```suggestion
/docs/guides/nginx https://0-20-0.docs.pomerium.com/docs/guides/nginx 301!
/docs/guides/traefik-ingress https://0-20-0.docs.pomerium.com/docs/guides/traefik-ingress 301!
```

Comment (desimone) [reply static/_redirects:280]: Agree; we should redirect for the latest deployed.

**Review by wasaga — APPROVED** [review]
redirect works

**Pull request merged**